src/script/script.hpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ /** @file script.hpp Base functions for all Scripts. */
```

Comment:
This comment is misleading, it's not some base fo all scripts but rather very narrow for event triggering of scripts.
I'd almost call it script_event.hpp, but that's also already in the api folder, and even though technically not a problem it doesn't seem like something worth in the long run.

Maybe 'script_trigger.hpp', and then 'Functionality to trigger events in AI and game scripts' as comment? Then I'd also rename Script to ScriptTrigger. It's not as concise, but it definitely is much less misleading.

Reply:
I renamed some things, the file, the Class name, and comment.